logprep/processor/decoder/decoders.py

```diff
@@ -0,0 +1,219 @@
+        r'"(?P<request_line>.*)"\s+'  # the requested document
+        r"(?P<status>\d{3})\s+"  # the http status returned to the client
+        r"(?P<bytes>\d+)\s*"  # content-length of the document transferred
+        r"$"
+    )
+)
+
+
+regex_nginx = (
+    re.compile(
+        r"^"
+        r"(?P<host>[^ ]*) - "
+        r"(?P<user>[^ ]*) "
+        r"\[(?P<time>[^\]]*)\]\s+"
+        r"(?P<code>[^ ]*) "
+        r'"(?P<method>\S+)(?: +(?P<path>[^\"]*?)(?: +\S*)?)?"\s+'
+        r"(?P<size>[^ ]*)\s+"
+        r'"(?P<referer>[^\"]*)"\s+'
+        r'"(?P<agent>[^\"]*)"\s+'
+        r'"(?P<gzip_ratio>[^\"]*)"'
+        r"$"
+    ),
+    re.compile(
+        r"^"
+        r"(?P<host>[^ ]*) - "
+        r"(?P<user>[^ ]*) "
+        r"\[(?P<time>[^\]]*)\]\s+"
+        r'"(?P<method>\S+)(?: +(?P<path>[^\"]*?)(?: +\S*)?)?"\s+'
+        r"(?P<code>[^ ]*) "
+        r"(?P<size>[^ ]*)\s+"
+        r'"(?P<referer>[^\"]*)"\s+'
+        r'"(?P<agent>[^\"]*)"'
+        r"$"
+    ),
+    re.compile(
+        r"^"
+        r"(?P<remote>[^ ]*) "
+        r"(?P<host>[^ ]*) "
+        r"(?P<user>[^ ]*) "
+        r"\[(?P<time>[^\]]*)\] "
+        r'"(?P<method>\S+)(?: +(?P<path>[^\"]*?)(?: +\S*)?)?" '
+        r"(?P<code>[^ ]*) "
+        r"(?P<size>[^ ]*)"
+        r'(?: "(?P<referer>[^\"]*)" '
+        r'"(?P<agent>[^\"]*)")'
+        r"$"
+    ),
+)
+
+
+regex_syslog_rfc3164_local = re.compile(  # local without host
+    r"^\<(?P<pri>[0-9]+)\>"
+    r"(?P<time>[^ ]* {1,2}[^ ]* [^ ]*) "
+    r"(?P<ident>[a-zA-Z0-9_\/\.\-]*)"
+    r"(?:\[(?P<pid>[0-9]+)\])?"
+    r"(?:[^\:]*\:)? *(?P<message>.*)"
+    r"$"
+)
+
+regex_syslog_rfc3164 = re.compile(
+    r"^\<(?P<pri>[0-9]+)\>"
+    r"(?P<time>[^ ]* {1,2}[^ ]* [^ ]*) "
+    r"(?P<host>[^ ]*) "
+    r"(?P<ident>[a-zA-Z0-9_\/\.\-]*)"
+    r"(?:\[(?P<pid>[0-9]+)\])?(?:[^\:]*\:)? "
+    r"*(?P<message>.*)"
+    r"$"
+)
+
+regex_iso8601 = r"\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)"
+
+regex_syslog_rfc5424 = re.compile(
+    r"^\<(?P<pri>[0-9]{1,5})\>"
+    r"1 "  # the version 1-> rfc5424
+    rf"(?P<time>{regex_iso8601}) "
+    r"(?P<host>[^ ]+) "
+    r"(?P<ident>[^ ]+) "
+    r"(?P<pid>[-0-9]+) "
+    r"(?P<msgid>[^ ]+) "
+    r"(?P<extradata>(\[(.*?)\]|-)) "
+    r"(?P<message>.+)$"
+)
+
+token_regex = re.compile(r'([a-zA-Z0-9]+)=("[^"]+"|\S+)')
+
+
+def parse_logfmt(log_line: str) -> dict[str, str]:
+    """parses logfmt format"""
+    tokens = token_regex.findall(log_line)
+    return {key: value.strip('"') for key, value in tokens}
+
+
+def parse_cri(log_line: str) -> dict[str, str]:
+    """parses cri container log format by using only
+    string operations"""
+    timestamp, _, log_line = log_line.partition(" ")
+    stream, _, log_line = log_line.partition(" ")
+    flags, _, message = log_line.partition(" ")
+    if any((not timestamp, not stream, not flags)):
+        raise DecoderError("can't be parsed with cri")
```

Comment:
Rather use an `all` than an `any not` construct. Even though we are handling strings I would find it more clear to use `len(x) > 0`

Reply:
turns out we have to use nothing of it, because the unpacking of `split` raises a `ValueError` if there are not enough Values to unpack. 😁